File: src/main/java/com/giffardtechnologies/restdocs/domain/Field.java
package com.giffardtechnologies.restdocs.domain;

public class Field extends TypeSpec {
	private String name;
	private String longName = "";
	private String description;
	private boolean required = true;

	public String getName() {
		return name;
	}
	
	public void setName(String name) {
		this.name = name;
	}

	public String getLongName() {
		return longName;
	}

	public void setLongName(String longName) {
		this.longName = longName;
	}

	public String getDescription() {
		return description;
	}
	
	public void setDescription(String description) {
		this.description = description;
	}
	
	public boolean isRequired() {
		return required;
	}
	
	public void setRequired(boolean required) {
		this.required = required;
	}

}

File: src/main/java/com/giffardtechnologies/restdocs/domain/Field.kt
package com.giffardtechnologies.restdocs.domain

import com.giffardtechnologies.restdocs.domain.type.TypeSpec
import io.vavr.collection.Array

data class Field(
    val name: String,
    val longName: String,
    val type: TypeSpec,
    val description: String? = null,
    val defaultValue: String? = null,
    val isRequired: Boolean = true,
    val sampleValues: Array<String> = Array.empty(),
    val parentName: String? = null,
) : FieldListElement
{
    val lazyContext by lazy { FieldContext(1) }

    var context: FieldContext = FieldContext(1)

    fun hasDefaultValue(): Boolean {
        return defaultValue != null
    }

    // for velocity templating
    val hasDefaultValue: Boolean
        get() = hasDefaultValue()

}

data class FieldContext(val dummy: Int?)